### PR TITLE
Show absolute time on timelines

### PR DESCRIPTION
**Pull request by kyori19**
Add an option in the settings to show the absolute time of every user’s post on timeline instead of relative time, as there frequently happens time delay if we use current system depending on refresh time of the timeline. This PR is made for those users who want to know the time by the default time stamp.
c23388134cd7af55
7f622e30147c61fd

Same as for now, we can use relative time displaying by disable this settings. See the attached image:
e3f677a4fd7ac0aa

### CODE REVIEW

**Comment by kyori19** [comment]
@connyduck 
I made some changes. Check please.

File: app/src/main/java/com/keylesspalace/tusky/adapter/NotificationsAdapter.java [hunk 1]
```diff
@@ -365,26 +377,40 @@ private void setUsername(String name) {
...
-            if (createdAt != null) {
-                long then = createdAt.getTime();
-                long now = new Date().getTime();
-                readout = DateUtils.getRelativeTimeSpanString(timestampInfo.getContext(), then, now);
-                readoutAloud = android.text.format.DateUtils.getRelativeTimeSpanString(then, now,
-                        android.text.format.DateUtils.SECOND_IN_MILLIS,
-                        android.text.format.DateUtils.FORMAT_ABBREV_RELATIVE);
+        protected void setCreatedAt(@Nullable Date createdAt) {
+            if (useAbsoluteTime) {
+                String time = "ERROR!";
```
Comment (connyduck) [inline app/src/main/java/com/keylesspalace/tusky/adapter/NotificationsAdapter.java]: the String doesnt need to be initialized here

File: app/src/main/java/com/keylesspalace/tusky/adapter/NotificationsAdapter.java [hunk 2]
```diff
@@ -365,26 +377,40 @@ private void setUsername(String name) {
...
+                if (createdAt != null) {
+                    SimpleDateFormat sdf;
+                    if (new Date().getTime() - createdAt.getTime() > 86400000L) {
+                        sdf = new SimpleDateFormat("MM/dd HH:mm:ss", Locale.getDefault());
+                    } else {
+                        sdf = new SimpleDateFormat("HH:mm:ss", Locale.getDefault());
```
Comment (connyduck) [inline app/src/main/java/com/keylesspalace/tusky/adapter/NotificationsAdapter.java]: make the `SimpleDateFormat`s class members then they dont need to be created for every timestamp. (Like the `BidiFormatter`)

File: app/src/main/java/com/keylesspalace/tusky/adapter/NotificationsAdapter.java [hunk 3]
```diff
@@ -365,26 +377,40 @@ private void setUsername(String name) {
...
+                    }
+                    time = sdf.format(createdAt);
+                    timestampInfo.setText(time);
+                }
```
Comment (connyduck) [inline app/src/main/java/com/keylesspalace/tusky/adapter/NotificationsAdapter.java]: there is no `else` branch for when the `createdAt` is null (just set the timestamp to ? or something)

File: app/src/main/java/com/keylesspalace/tusky/adapter/StatusBaseViewHolder.java [hunk 4]
```diff
@@ -126,25 +132,39 @@ void setAvatar(String url, @Nullable String rebloggedUrl) {
     }
 
     protected void setCreatedAt(@Nullable Date createdAt) {
```
Comment (connyduck) [inline app/src/main/java/com/keylesspalace/tusky/adapter/StatusBaseViewHolder.java]: same as in `NotificationsAdapter`

**Comment by kyori19** [comment]
@connyduck Fixed problems.

**Comment by connyduck** [comment]
Almost! Use two `SimpleDateFormat` members and remove the `passed1day` and the it should be perfect. 
pseudocode:
```
private SimpleDateFormat sameDayFormat
private SimpleDateFormat differentDayFormat   

....

if (new Date().getTime() - createdAt.getTime() > 86400000L) {
   time = differentDayFormat.format(createdAt);
} else {
   time = sameDayFormat.format(createdAt)
    }
```

also thinking of it `System.currentTimeMillis()` should be marginally faster than `new Date().getTime()` because it doesnt create a new object.

**Comment by kyori19** [comment]
@connyduck I could build apk on my PC but CircleCI was failed...
Please retry it...

**Comment by connyduck** [comment]
Ok now the code looks fine ^__^
I will merge it later for Tusky 4.
Don't worry about CircleCI.

**Pull request merged**